cloud-gateway/src/main/java/com/registry/cloudgateway/controller/CloudGatewayController.java

```diff
@@ -22,28 +25,35 @@ public ClientDTO getClientById(@PathVariable("id") String clientNumber) {
     }
 
     @GetMapping("/clients")
-    public List<ClientDTO> getAllClients() {
-        return cloudGatewayService.getAllClients();
+    public ResponseEntity<RestResponsePage<ClientDTO>> getAllClients(@RequestParam(value = "page", required = false) Integer page,
+                                                                     @RequestParam(value = "size", required = false) Integer size) {
+        return cloudGatewayService.getAllClients(page, size);
     }
 
-    @PostMapping("clients")
+    @PostMapping("/clients")
     public ClientDTO createClient(@RequestBody ClientDTO clientDTO) {
         return cloudGatewayService.createClient(clientDTO);
     }
 
     @GetMapping("/invoices/{id}")
-    public InvoiceDTO getInvoiceById(@PathVariable("id") Integer id) {
+    public InvoiceDTO getInvoiceById(@PathVariable("id") Long id) {
         return cloudGatewayService.getInvoiceById(id);
     }
 
     @GetMapping("/invoices")
-    public List<InvoiceDTO> getAllInvoices() {
-        return cloudGatewayService.getAllInvoices();
+    public ResponseEntity<RestResponsePage<InvoiceDTO>> getAllInvoices(@RequestParam(value = "page", required = false) Integer page,
+                                                                       @RequestParam(value = "size", required = false) Integer size) {
+        return cloudGatewayService.getAllInvoices(page, size);
     }
 
     @PostMapping("/invoices")
     public InvoiceDTO createInvoice(@RequestBody InvoiceDTO invoiceDTO) {
         return cloudGatewayService.createInvoice(invoiceDTO);
     }
 
+    @PostMapping("/invoices/search")
+    public List<InvoiceDTO> searchInvoice(@RequestBody CriteriaDTO criteriaDTO) {
```

Comment:
I would call this DTO something less generic. For example InvoiceSearchRequestDTO.